person.rb

```diff
@@ -23,5 +26,37 @@ def can_use_services?
     end
   end
 
+  def correct_name
+    @name
+  end
+
   private :is_of_age?
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    name = @nameable.correct_name
+    if name.length > 10
+      name[0...10]
```

Comment:
- hey, you want the name to be 10 characters and not 11 characters, as this will return 11 characters. all you need is [0...9] as it will return 10 characters. let's fix this.